src/Microsoft.Azure.WebJobs.Host/Loggers/FunctionInstanceLogger.cs

```diff
@@ -24,12 +24,17 @@ public FunctionInstanceLogger(ILoggerFactory loggerFactory)
 
         public Task<string> LogFunctionStartedAsync(FunctionStartedMessage message, CancellationToken cancellationToken)
         {
-            string traceMessage = string.Format(CultureInfo.InvariantCulture, "Executing '{0}' (Reason='{1}', Id={2})", message.Function.ShortName, message.FormatReason(), message.FunctionInstanceId);
-            Log(LogLevel.Information, message.Function, message.FunctionInstanceId, traceMessage);
+            string name = message.Function.ShortName;
+            string reason = message.ReasonDetails ?? message.FormatReason();
+            var id = message.FunctionInstanceId;
+
+            Log(LogLevel.Information, message.Function, (s, e) => $"Executing '{name}' (Reason='{reason}', Id={id})");
```

Comment:
Not really an expensive log statement, but happens on every request, so I've moved it to the formatter func pattern

Reply:
Did you look at changing this to the `LoggerMessage.Define` pattern at all? https://github.com/Azure/azure-webjobs-sdk/pull/2235.

It looks like there's some allocation savings doing it this way (esp for such an often-called message) -- but I clearly dropped the ball and never pushed this over the finish line.

Reply:
I've changed the executing/executed logs to use a defined delegate. I've also added logger caching.

I've left the templatized event details log as is for now, since that's more complicated. We can address that later. That's not logged on http functions anyways.